pento/free-as-in-speech
# Add tool to anonymize HAR files

As mentioned in our [Working with the Command-line](https://github.com/pento/free-as-in-speech/blob/anonymize-har/docs/contributors/command-line.md) docs, HAR files can contain PII (Personally Identifiable Information) which we want to avoid.

Thus, this implements a tool that can rewrite HAR files to be more anonymous. Since it contains a list of rules, there might still be some rules missing so that it needs to be reviewed from time to time.

## Testing instructions

Run the script on a HAR file:
```
$ node packages/fetch-from-har/bin/anonymize-har.js test.har
Reading test.har...

The following number of items were replaced:
 - 762 timestamp
 - 13 ipv4
 - 5 ipv6
 - 78 email
 - 948 uuid
 - 1 wixSite
 - 1 _wixUIDX
 - 1 mediaToken
 - 1 wixSession2
 - 10 authorization
 - 3 session_id
 - 176 request_id
Wrote anonymized-test.har.
```

Then verify that the anonymized file contains changes and that it still works fine:
```
$ diff test.har anonymized-test.har | wc -l
   21608
$ diff test.har anonymized-test.har | $EDITOR
$ open anonymized-test.har|
```

Fixes #24
